pkg/monitoring/alloy/service.go

```diff
@@ -35,9 +36,16 @@ func (a *Service) ReconcileCreate(ctx context.Context, cluster *clusterv1.Cluste
 	logger := log.FromContext(ctx)
 	logger.Info("alloy-service - ensuring alloy is configured")
 
+	// Get list of tenants
+	var tenants []string
+	tenants, err := commonmonitoring.ListTenants(a.Client, ctx)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
 	configmap := ConfigMap(cluster)
-	_, err := controllerutil.CreateOrUpdate(ctx, a.Client, configmap, func() error {
-		data, err := a.GenerateAlloyMonitoringConfigMapData(ctx, configmap, cluster)
+	_, err = controllerutil.CreateOrUpdate(ctx, a.Client, configmap, func() error {
+		data, err := a.GenerateAlloyMonitoringConfigMapData(ctx, configmap, cluster, tenants)
 		if err != nil {
```

Comment:
Why not add the function to the alloy service if that's the only place we use it at?

Reply:
That's what I did at first but then I thought it would be cleaner to have it in the common package.

Reply:
Updated